src/simulator/data_gen.c

```diff
@@ -95,46 +77,35 @@ data_gen_sample_create(DataGen* data, u32 idx) {
 }
 
 
-internal void
-data_gen_sample_destroy(DataSample* sample) {
-    check(sample != NULL, "sample is NULL");
-    
-    memset(sample, 0, sizeof(*sample));
-    memory_free(sample);
-    
-    error:
-    return;
-}
-
-
 /***********************
 * NETWORK INPUTS
 ***********************/
 internal NetworkInputs*
-data_network_inputs_create(DataSample* sample, Network* network, u32 time) {
+data_network_inputs_create(State* state, DataSample* sample, Network* network, u32 time) {
```

Comment:
Maybe only give the arena?, check if that can be done and only give the arena for transient to avoid bugs

Reply:
Done